src/output-json-anomaly.c

```diff
@@ -348,9 +361,29 @@ static void SetFlag(const ConfNode *conf, const char *name, uint16_t flag, uint1
 static void JsonAnomalyLogConf(AnomalyJsonOutputCtx *json_output_ctx,
         ConfNode *conf)
 {
-    uint16_t flags = 0;
+    static bool _warn_no_flags = false;
+    static bool _warn_no_packet = false;
+    uint16_t flags = ANOMALY_DEFAULTS;
     if (conf != NULL) {
-        SetFlag(conf, "packethdr", LOG_JSON_PACKETHDR, &flags);
+        /* Check for metadata to enable/disable. */
+        ConfNode *typeconf = ConfNodeLookupChild(conf, "types");
+        if (typeconf != NULL) {
+            SetFlag(typeconf, "application-layer", LOG_JSON_APPPROTO_TYPE, &flags);
+            SetFlag(typeconf, "packet", LOG_JSON_PACKET_TYPE, &flags);
+        }
+        SetFlag(conf, "packethdr",   LOG_JSON_PACKETHDR,   &flags);
+    }
+    if (((flags & (LOG_JSON_PACKET_TYPE | LOG_JSON_PACKETHDR)) == LOG_JSON_PACKETHDR) && !_warn_no_packet) {
+        SCLogNotice("Anomaly logging configured to include packet headers, however packet "
```

Comment:
I think this should be a SCLogWarning instead.

Reply:
Fixed.